File: src/main/java/loxal/epvin/datastore/model/DatastoreEntity.java
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.model;

import com.googlecode.objectify.annotation.Entity;
import com.googlecode.objectify.annotation.Id;
import com.googlecode.objectify.annotation.OnSave;


/**
 * @author Alexander Orlov <alexander.orlov@loxal.net>
 */
@Entity
public class DatastoreEntity {
    @OnSave
    void prePersist() {
        this.version++;
    }

    @Id
    private Long id;

    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    private Integer version = 0;

    public Integer getVersion() {
        return version;
    }

    public void setVersion(final Integer version) {
        this.version = version;
    }
}

File: src/main/kotlin/loxal/epvin/datastore/model/DatastoreEntity.kt
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.model

import com.googlecode.objectify.annotation.Entity
import com.googlecode.objectify.annotation.Id
import com.googlecode.objectify.annotation.OnSave

Entity
public open class DatastoreEntity {
    OnSave
    fun prePersist() {
        this.version++
    }

    Id
    public var id: Long? = null

    public var version: Int = 0
}